packages/graphql/tests/integration/directives/cypher/cypher-with-non-node-types.int.test.ts

```diff
@@ -0,0 +1,316 @@
+            `;
+
+        const gqlResult = await testHelper.executeGraphQL(query);
+
+        expect(gqlResult.errors).toBeUndefined();
+        expect(gqlResult.data).toEqual({
+            [Movie.plural]: [
+                {
+                    title: "The Matrix",
+                    links: expect.toIncludeSameMembers([
+                        { name: "Main Website", url: "the-matrix.org" },
+                        { name: "Public Database", url: "not-imdb.com" },
+                    ]),
+                },
+            ],
+        });
+    });
+
+    test("nested field", async () => {
+        const Movie = testHelper.createUniqueType("Movie");
+        const Actor = testHelper.createUniqueType("Actor");
+        const Link = testHelper.createUniqueType("Link");
+
+        const typeDefs = /* GraphQL */ `
+            type ${Actor} @node {
+                name: String
+                movies: [${Movie}!]!  @relationship(type: "ACTED_IN", direction: OUT)
+            }
+
+            type  ${Movie} @node {
+               title: String
+               id: String!
+               link: ${Link}! @cypher(statement: """
+               MATCH(l:${Link})
+               WHERE l.movieId=this.id
+               RETURN l {.name, .url} as link
+               """, columnName: "link")
+            }
+
+            type ${Link} {
+                movieId: String!
+                url: String!
+                name: String!
+            }
+        `;
+
+        await testHelper.initNeo4jGraphQL({ typeDefs });
+
+        await testHelper.executeCypher(`
+            CREATE (:${Actor} {name: "Keanu"})-[:ACTED_IN]->(:${Movie} { id: "matrix", title: "The Matrix"})
+            CREATE (:${Link} {movieId: "matrix", url: "the-matrix.org", name: "Main Website"})
+        `);
+
+        const query = /* GraphQL */ `
+                {
+                    ${Actor.plural} {
+                        name
+                        movies {
+                            title
+                            link {
+                                url
+                                name
+                            }
+                        }
+                    }
+                }
+            `;
+
+        const gqlResult = await testHelper.executeGraphQL(query);
+
+        expect(gqlResult.errors).toBeUndefined();
+        expect(gqlResult.data).toEqual({
+            [Actor.plural]: [
+                {
+                    name: "Keanu",
+                    movies: [
+                        {
+                            title: "The Matrix",
+                            link: { name: "Main Website", url: "the-matrix.org" },
+                        },
+                    ],
+                },
+            ],
+        });
+    });
+
+    /**
+     * Interfaces without `@node` are not supported yet because there is no way to access "__resolveTree", as the information of the concrete type returned by the cypher field is not available
+     */
```

Comment:
What will happen in this case? Will we throw a validation error? Or just some weird indecipherable error?